AK/Types.h

```diff
@@ -84,6 +84,41 @@ namespace AK_REPLACED_STD_NAMESPACE { // NOLINT(cert-dcl58-cpp) nullptr_t must b
 using nullptr_t = decltype(nullptr);
 }
 
+#if !USING_AK_GLOBALLY || defined(AK_DONT_REPLACE_STD)
+// initializer_list must be in ::std:: in order to work, so it is impossible to use AK_REPLACED_STD_NAMESPACE trick here.
+#    include <initializer_list>
+#else
+namespace std {
+// 17.10.2 Header <initializer_list> synopsis
```

Comment:
Not a fan of this change tbh. This definition is specific to each compiler. They tend to get very specific. 

If we go down this road, we end up with this header when trying to use coroutines, which is just nasty:

<https://github.com/SerenityOS/serenity/commit/1b1685b1f5389ddfdf4da7931bca4e60700e7b5a>

This kind of stuff is also the reason why we don't use `__builtin_source_location()` in `AK::SourceLocation`. The GCC frontend checks that the builtin is only called within a struct that has all the exact properties of the libstdc++ `std::source_location` (or at least, it did for gcc ~11).

Reply:
Dropped the change.